arch/src/aarch64/fdt.rs

```diff
@@ -81,6 +89,119 @@ pub enum Error {
+/// L1 instrcution cache
+const L1_I_CACHE: u32 = 1;
+/// L2 cache
+const L2_CACHE: u32 = 2;
+/// L3 cache
+const L3_CACHE: u32 = 3;
+
+///let mut file_directory = "/sys/devices/system/cpu/cpu0/cache/index0/size";
+pub fn get_cache_size(cache_level: u32) -> u32 {
+    let mut file_directory: String = "/sys/devices/system/cpu/cpu0/cache".to_string();
+    if cache_level == L1_D_CACHE {
+        file_directory += "/index0/size";
+    } else if cache_level == L1_I_CACHE {
+        file_directory += "/index1/size";
+    } else if cache_level == L2_CACHE {
+        file_directory += "/index2/size";
+    } else if cache_level == L3_CACHE {
+        file_directory += "/index3/size";
+    }
+
+    let mut cache_size: u32 = 0;
+    let file_path = Path::new(&file_directory);
+    let file_exist: bool = file_path.exists();
+    if !file_exist {
+        error!("File: {} not exist.", file_directory);
+    } else {
+        info!("File: {} exist.", file_directory);
+
+        let src0 = fs::read_to_string(file_directory).expect("File not exists or file corrupted.");
+        let src = src0.trim();
+        let src_l = &src[0..src.len() - 1];
+        let src_r = &src[src.len() - 1..];
+
+        let size_k: &str = "K";
+        let size_m: &str = "M";
+        let size_g: &str = "G";
+
+        //Add the cpu cache info
+        let temp: u32 = src_l.parse().unwrap();
+        if src_r == size_k {
+            cache_size = temp * 1024;
+        } else if src_r == size_m {
+            cache_size = temp * 1024 * 1024;
+        } else if src_r == size_g {
+            cache_size = temp * 1024 * 1024 * 1024;
+        } else {
+            cache_size = temp;
+        }
+    }
+
+    cache_size
+}
+
+///let file_directory = "/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size";
+pub fn get_cache_coherency_line_size(cache_level: u32) -> u32 {
+    let mut file_directory: String = "/sys/devices/system/cpu/cpu0/cache".to_string();
+    if cache_level == L1_D_CACHE {
+        file_directory += "/index0/coherency_line_size";
+    } else if cache_level == L1_I_CACHE {
+        file_directory += "/index1/coherency_line_size";
+    } else if cache_level == L2_CACHE {
+        file_directory += "/index2/coherency_line_size";
+    } else if cache_level == L3_CACHE {
+        file_directory += "/index3/coherency_line_size";
+    }
+
+    let mut cache_coherency_line_size: u32 = 0;
+    let file_path = Path::new(&file_directory);
+    let file_exist: bool = file_path.exists();
+    if !file_exist {
+        error!("File: {} not exist.", file_directory);
+    } else {
+        info!("File: {} not exist.", file_directory);
+
+        let src0 = fs::read_to_string(file_directory).expect("File not exists or file corrupted.");
+        let src = src0.trim();
+        cache_coherency_line_size = src.parse().unwrap();
+    }
+
+    cache_coherency_line_size
+}
+
+///let file_directory = "/sys/devices/system/cpu/cpu0/cache/index0/number_of_sets";
```

Comment:
Commeted out code?

Reply:
It is not commented out code, just annotation. I will modify it soon.